src/hypercall.rs

```diff
@@ -77,21 +79,77 @@ pub unsafe fn address_to_hypercall(
 }
 
 /// unlink deletes a name from the filesystem. This is used to handle `unlink` syscalls from the guest.
-/// TODO: UNSAFE AS *%@#. It has to be checked that the VM is allowed to unlink that file!
-pub fn unlink(mem: &MmapMemory, sysunlink: &mut UnlinkParams) {
-	unsafe {
-		sysunlink.ret = libc::unlink(mem.host_address(sysunlink.name).unwrap() as *const i8);
+pub fn unlink(mem: &MmapMemory, sysunlink: &mut UnlinkParams, file_map: &mut UhyveFileMap) {
+	// For more information on how this works, check out the open call implementation.
+	let requested_path = mem.host_address(sysunlink.name).unwrap() as *const i8;
+	let guest_path = unsafe { CStr::from_ptr(requested_path) }.to_str();
+	if let Ok(guest_path) = guest_path {
+		if let Some(host_path) = file_map.get_host_path(guest_path) {
+			let host_path_c_string = CString::new(host_path.as_bytes()).unwrap();
+			unsafe {
+				sysunlink.ret = libc::unlink(host_path_c_string.as_c_str().as_ptr());
+			}
+		} else {
+			error!("The kernel requested to unlink() an unknown path ({guest_path}): Rejecting...");
+			sysunlink.ret = -1;
```

Comment:
Would be nice to have the `-1` as a const in _uhyve_interface_

Reply:
I worked around that by using proper errors here: https://github.com/hermit-os/uhyve/pull/783/commits/0cf67bae2f5a8dbe284884d820b727420751a9df